docs/mkdocs/preflight/preflight-validation.md

```diff
@@ -0,0 +1,100 @@
+# Preflight validition of the cluster with installed KMM Module
+
+Before executing upgrade on the cluster with applied KMM Modules, admin needs to verify that installed kernel modules (via KMM)
+will be able to be installed on the nodes after the cluster upgrade and possible kernel upgrade. 
+Preflight will try to validate every Module loaded in the cluster, in parallel (it does not wait for validation of one Module to complete, before starting
+validation of another Module).
+
+### Validation kick-off
+Preflight validation is triggered by uploading PreflightValidation CR into the cluster. This CR's Spec contains 2 fields:
+```
+type PreflightValidationSpec struct {
+        // KernelImage describes the kernel image that all Modules need to be checked against.
```

Comment:
Nit: We should probably update the doc to fit the `KernelVersion` instead of `KernelImage` in a later PR.